src/backend/PostgresBackend.cpp

```diff
@@ -393,153 +450,99 @@ PostgresBackend::doFetchSuccessorKey(
-        BOOST_LOG_TRIVIAL(info)
-            << __func__ << " fetched " << std::to_string(hashes.size())
-            << " transactions with threadpool. took "
-            << std::to_string(duration);
+    PgQuery pgQuery(pgPool_);
+    pgQuery("SET statement_timeout TO 10000", yield);
+    std::stringstream sql;
+    for (size_t i = 0; i < hashes.size(); ++i)
+    {
+        auto const& hash = hashes[i];
+        sql << "SELECT transaction,metadata,ledger_seq,date FROM "
+               "transactions "
+               "WHERE HASH = \'\\x"
+            << ripple::strHex(hash) << "\'";
+        if (i + 1 < hashes.size())
+            sql << " UNION ALL ";
     }
-    else
+    auto start = std::chrono::system_clock::now();
+    auto res = pgQuery(sql.str().data(), yield);
+    auto end = std::chrono::system_clock::now();
+    auto duration = ((end - start).count()) / 1000000000.0;
+    BOOST_LOG_TRIVIAL(info)
+        << __func__ << " fetched " << std::to_string(hashes.size())
+        << " transactions with union all. took " << std::to_string(duration);
+    if (size_t numRows = checkResult(res, 3))
     {
-        PgQuery pgQuery(pgPool_);
-        pgQuery("SET statement_timeout TO 10000");
-        std::stringstream sql;
-        for (size_t i = 0; i < hashes.size(); ++i)
-        {
-            auto const& hash = hashes[i];
-            sql << "SELECT transaction,metadata,ledger_seq,date FROM "
-                   "transactions "
-                   "WHERE HASH = \'\\x"
-                << ripple::strHex(hash) << "\'";
-            if (i + 1 < hashes.size())
-                sql << " UNION ALL ";
-        }
-        auto start = std::chrono::system_clock::now();
-        auto res = pgQuery(sql.str().data());
-        auto end = std::chrono::system_clock::now();
-        auto duration = ((end - start).count()) / 1000000000.0;
-        BOOST_LOG_TRIVIAL(info)
-            << __func__ << " fetched " << std::to_string(hashes.size())
-            << " transactions with union all. took "
-            << std::to_string(duration);
-        if (size_t numRows = checkResult(res, 3))
-        {
-            for (size_t i = 0; i < numRows; ++i)
-                results.push_back(
-                    {res.asUnHexedBlob(i, 0),
-                     res.asUnHexedBlob(i, 1),
-                     res.asBigInt(i, 2),
-                     res.asBigInt(i, 3)});
-        }
+        for (size_t i = 0; i < numRows; ++i)
+            results.push_back(
+                {res.asUnHexedBlob(i, 0),
+                 res.asUnHexedBlob(i, 1),
+                 res.asBigInt(i, 2),
+                 res.asBigInt(i, 3)});
     }
+
     return results;
 }
 
 std::vector<Blob>
 PostgresBackend::doFetchLedgerObjects(
     std::vector<ripple::uint256> const& keys,
-    uint32_t sequence) const
+    uint32_t sequence,
+    boost::asio::yield_context& yield) const
 {
+    if (!keys.size())
+        return {};
+
+    std::vector<Blob> results = {};
```

Comment:
If we're anticipating that `results` will contain roughly `keys.size()` entries, consider increasing the vector's capacity with `reserve`.